aspnetcore-webapi/GettingStartedASPNETMongoDB/GettingStartedASPNETMongoDB/Models/Course.cs

```diff
@@ -0,0 +1,18 @@
+﻿using MongoDB.Bson;
+using MongoDB.Bson.Serialization.Attributes;
+using System.ComponentModel.DataAnnotations;
+
+namespace GettingStartedASPNETMongoDB.Models;
+
+public class Course
+{
+    [BsonId]
+    [BsonRepresentation(BsonType.ObjectId)]
+    public string? Id { get; set; }
+
+    [Required(ErrorMessage = "Course name is required")]
+    public string Name { get; set; }
+
+    [Required(ErrorMessage = "Course code is required")]
+    public string Code { get; set; }
```

Comment:
Properties that can be null should be marked as nullable, and properties that are not nullable, they should be set to null! .